apps/target_manager.py

```diff
@@ -41,6 +51,33 @@ def create_target(targets_json, compose_apps, ota_lite_tag, git_sha, in_version=
     new_targets = {}
     for tag, latest in latest_targets.items():
         for target in latest.values():
+            # Assumption/Policy is that the latest version should be same across all hwids/machines for a specific tag,
+            # if it's not true for some parent Target then we skip creation of the new Target. It can be so
+            # in the case if the expected version of the parent Target failed and for some reason we don't
+            # want to create a new Target based on the latest available Target
+            if int(target['custom']['version']) != latest_tag_version[tag]:
+                logger.warning('Skipping Target creation: {}, corresponding latest parent Target has not been found'
+                               .format(tagmgr.create_target_name(target, version, tag),
+                                       tagmgr.create_target_name(target, str(latest_tag_version[tag]), tag)))
+                continue
+
+            # TODO: refactor it in order to bring to common denominator with the other parts of code
```

Comment:
Can you put a comment on what you are checking for here? I'm having a hard time tracking this. Would splitting this up into some helper function make it any cleaner/clearer?

Reply:
>Can you put a comment on what you are checking for here?
Yes, I'll add a comment, effectively we discussed it last Thursday.

>  Would splitting this up into some helper function make it any cleaner/clearer?
Yes, makes sense, I'll try to make the code cleaner and more understandable.

Reply:
Restructured the code a little bit and added more comments along with the use cases' examples.